### ISSUE TITLE
Add Archive action to lists (All Mail, Basic Search, Advanced Search, etc.)

### ISSUE THREAD

**Issue by marclaporte** [jasonmunro/cypht#439]
We have https://github.com/jasonmunro/cypht/issues/228   Great!

![2020-10-06_152437](https://user-images.githubusercontent.com/1004261/95250739-829bfc80-07e8-11eb-889f-001c97c5ee2d.png)

Please keep in mind that archiving acts differently in each account (it is configurable) so if we are in "All Mail" interface, each mail needs to be archived as if we launched the action directly on the email.

**Comment by jasonmunro** [comment]
PR for this has been merged so pretty sure we can close this out @marclaporte

**Comment by marclaporte** [comment]
Testing with latest Cypht from Packagist bundled in Tiki:
I see the buttons in advanced search and all mail, but clicking on it does nothing. Perhaps I have some sticky cache?

**Comment by jasonmunro** [comment]
I actually did not test the archive process. Do you have archive folders setup for your accounts?

**Comment by marclaporte** [comment]
When archiving from an email view, Cypht uses a default pattern (which I am OK with) and it creates folders as needed.

So I usually do not set anything. I expect archive from search or list to work the same way.

**Comment by jasonmunro** [comment]
I respectfully disagree about that. An archive folder is IMO like a special folder (sent/trash/drafts) and auto-creating those is I think a presumptuous action for a webmail client. Not to say we could not auto-create one via the nux module or a different avenue, but at the time of an archive action is IMO too late. Auto-creating also makes some assumptions I don't think we want to make (i18n for example).

**Comment by marclaporte** [comment]
I see what you mean. The current behavior is not what I was expecting, but that is what @yvesdaxmaz coded and I felt it was good enough (and we could always improve later). The archive discussion was here: https://github.com/jasonmunro/cypht/issues/228

I use "Archive" all the time. But other users may prefer to only have the archive button if the archive feature is set up. But then, we'll get feedback from @dumblob about how buttons should not change :-)

So @jasonmunro, how shall we proceed now?

**Comment by marclaporte** [comment]
Thank you @jasonmunro for https://github.com/jasonmunro/cypht/commit/268ec8a5d8c1f8a2d9296e2a5cb5907b476434d9

I will test.

**Comment by marclaporte** [comment]
When I try to archive a few messages from a list, I get "An error occurred trying to archive some messages!"

**Comment by jasonmunro** [comment]
looks like we still have issues, and the auto-create if it does not exist is still present. I'm going to remove that today and clean up the rest. I will tie it to this issue when I push the fixes.

**Comment by marclaporte** [comment]
With latest code: Nice error message if Archive folder is not set. And it then works on one message.

But does not work in list mode where I get "An error occurred trying to archive some messages!"

**Comment by jasonmunro** [comment]
works fine here so I'm not sure what is happening. Likely need to log the IMAP debug info

**Comment by marclaporte** [comment]
Ok, I think that it's because Cypht in Tiki is not reliably saving the archive folder setting. So I will close as done, and re-open if relevant.

Thanks!

**Comment by Steven--** [comment]
I'm having the same problem as @marclaporte on a fresh Cypht install on Debian 11. 
Using the debug mode does not seem to produce any useful messages on the error log. What kind of info can I provide to help troubleshoot the issue?

Also, this is probably a dumb question, but how do I check the version of the installed Cypht? I don't see it reported anywhere.

**Comment by marclaporte** [comment]
@Steven-- I'll re-open this.

Can you join us on https://gitter.im/cypht-org/community so we can troubleshoot?

Thanks!

**Comment by Steven--** [comment]
@marclaporte Sure! I'm a bit busy at the moment but I'll join in a few days. Thank you.
